src/kakarot/kakarot.cairo

```diff
@@ -438,3 +447,22 @@ func eth_send_transaction{
 
     return result;
 }
+
+@l1_handler
+func handle_l1_message{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}(from_address: felt, l1_sender: felt, to_address: felt, value: felt, data_len: felt, data: felt*) {
+    alloc_locals;
+    let is_authorized = Kakarot.get_authorized_message_sender(from_address);
```

Comment:
I don't understand this authorization need, who is supposed to be authorized?

Reply:
Only the L1KakarotMessaging contract should be authorized to send messages, to prevent others from sending messages and spoofing the l1_sender.

Reply:
Only the L1KakarotMessaging contract should be authorized to send messages, to prevent others from sending messages and spoofing the l1_sender.

Reply:
Only the L1KakarotMessaging contract should be authorized to send messages, to prevent others from sending messages and spoofing the l1_sender.